internal/annotation/pause.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotation
+
+// PausedKey being present on an object with a truthy value
+// results in that object not passing a predicate or being added to a queue.
+// Read: this maps to a falsy filter (think truthy complement).
+const PausedKey = "operatorframework.io/paused"
```

Comment:
Out of curiousity, for non-sdk namespaced annotations (which I assume would be like `sdk.operatorframework.io` or something), is there any place where these more global operatorframework annotations would be tracked? 100% not a blocker. just curious if there's any risk that another project like OLM might want to add something similar but perhaps with slightly different semantics and not realize that we already have claimed this annotation?

Reply:
There is none currently. I'd expect that would be documented in the api repo under some `docs/annotations/<repo or project or component>.md` file. Great call-out.

Reply:
Alternatively, we could just make the annotation key an input into the library to skirt this problem.

From an operator author perspective, I would rather not leak implementation details (like that I'm using operatorframework) to end users. Instead, I could use `my.domain/paused`.

Reply:
https://github.com/operator-framework/api/issues/122

Reply:
>From an operator author perspective, I would rather not leak implementation details (like that I'm using operatorframework) to end users. Instead, I could use my.domain/paused

Right I do see this as a concern, but does having non-standard namespacing break [annotation conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set)? Also is it really that much more of a leap to infer that if `my.domain/paused` is set that a controller is using operator-lib's pause mechanism? The relationship becomes implicit but is still there via this repo.

Reply:
>We'd be delegating annotation convention adherence to the operator author, right? Or we could potentially validate that the provided string follows the convention.

I'm more concerned with semantics rather than format, from the angle of "operator-framework owns the pause mechanism, and so the annotation should be namespaced as such". However I am amenable to the argument that just because you're using a library does not mean you must delegate ownership of some process to that library (unless ownership delegation is required by library contract, which it does not have to be here).

>On the flipside, if an end user is able to determine that a project is using operator framework, will they assume that they can use this annotation?

I think this is a potential security issue actually. What if a user can pause their CR themselves? This annotation should be applicable by admins only.

>end users should go to operator docs (not operatorframework docs) to determine what annotations are supported by an operator and what each of those annotations does.

In general I agree, for non-admin annotations. But for admin-only annotations like this pause feature, the meaning of an annotation should be opaque and no docs should exist for end users.

Reply:
> I think this is a potential security issue actually. What if a user can pause their CR themselves? This annotation should be applicable by admins only.

The `metadata.annotations` field is not separable from the rest of the custom object from an RBAC perspective, so the only way to restrict a particular annotation from being set would be to introduce an admission webhook, right?

Reply:
Yeah, so this is out of scope here but should be noted in the docs. Is there a typical way one would write an admission controller to check which RBAC subject is modifying an object?

Reply:
The subject is included in the admission request object. Docs here: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#request